Mathlib/SetTheory/Surreal/Division.lean

```diff
@@ -0,0 +1,714 @@
+      Sum.elim_inr, exists_const, Sum.forall, le_refl, or_true, zero_rightMoves, IsEmpty.exists_iff,
+      or_false, implies_true, and_true, rightMoves_mk, moveRight_mk]
+  )
+  · constructor
+    · intro i
+      left
+      by_contra bad
+      simp only [not_or, not_exists, not_and, PGame.not_le] at bad
+      rw [lf_iff_lt numeric_zero (x_num.2.1 i)] at bad
+      rcases bad with ⟨bad1, bad2⟩
+      specialize bad1 i bad2
+      exact lf_irrefl _ bad1
+    · intro j
+      right
+      use j
+  · constructor
+    · intro i
+      left
+      left
+      use i
+    · intro j
+      right
+      use j
+
+/-! ### Options of inv' are numeric
+
+  We prove by induction that if `x` is numeric, then the options of `x.inv'` are numeric. Note that
+  this does *not* prove that `x.inv'` is numeric, since we have not proven the other condition of
+  numericity, that all Left options are less than all Right options. This other condition is
+  proven later in `onag_1_10_ii`.
+-/
+
+lemma invVal_numeric {l r} {L : l → PGame} {R : r → PGame}
+    (h3 : ∀ i, 0 < L i → (L i).inv'.Numeric)
+    (h4 : ∀ j, (R j).inv'.Numeric) (h5 : (PGame.mk l r L R).Numeric) {b : Bool}
+    (i : InvTy {i // 0 < L i} r b) :
+      (invVal (fun (i : {i // 0 < L i}) => L i) R (fun i => (L i).inv') (fun j => (R j).inv')
+        (PGame.mk l r L R) i).Numeric := by
+  induction i <;> simp only [invVal]
+  · exact numeric_zero
+  all_goals
+    case _ i j ih =>
+      apply Numeric.mul
+      · apply Numeric.add
+        · exact numeric_one
+        · apply Numeric.mul
+          · apply Numeric.sub
+            · first | exact h5.2.1 i | exact h5.2.2 i
+            · exact h5
+          · exact ih
+      · first | exact h4 i | exact h3 i i.2
+
+lemma inv'_numeric (ih_xl : ∀ i, 0 < x.moveLeft i → (x.moveLeft i).inv'.Numeric)
+    (ih_xr : ∀ j, (x.moveRight j).inv'.Numeric) :
+    (∀ i, (x.inv'.moveLeft i).Numeric) ∧ (∀ j, (x.inv'.moveRight j).Numeric) := by
+  constructor <;> (
+    intro i
+    rcases x with ⟨xl, xr, xL, xR⟩
+    exact invVal_numeric ih_xl ih_xr x_num i
+  )
+
+lemma inv'_numeric_left (ih_xl : ∀ i, 0 < x.moveLeft i → (x.moveLeft i).inv'.Numeric)
+    (ih_xr : ∀ j, (x.moveRight j).inv'.Numeric) :
+    ∀ i, (x.inv'.moveLeft i).Numeric := (inv'_numeric x_num x_pos ih_xl ih_xr).1
+
+lemma inv'_numeric_right (ih_xl : ∀ i, 0 < x.moveLeft i → (x.moveLeft i).inv'.Numeric)
+    (ih_xr : ∀ j, (x.moveRight j).inv'.Numeric) :
+    ∀ j, (x.inv'.moveRight j).Numeric := (inv'_numeric x_num x_pos ih_xl ih_xr).2
+
+/-! ### Helper lemmas for the main proof
+
+  These lemmas do the heavy lifting within the proof of the main result `onag_1_10`.
+  They are stated with additional hypotheses which will be satisfied by induction
+  during the main proof.
+
+-/
+
+/-- Given `y''`, return `(x', x'_inv, y')` -/
+def components {l r} {L : l → PGame} {R : r → PGame}
+    (h3 : ∀ i, 0 < L i → (L i).inv'.Numeric)
+    (h4 : ∀ j, (R j).inv'.Numeric) (h5 : (PGame.mk l r L R).Numeric) {b : Bool}
```

Comment:
Maybe something like this?
```suggestion
    (hnum_L : ∀ i, 0 < L i → (L i).inv'.Numeric)
    (hnum_R : ∀ j, (R j).inv'.Numeric) (hnum : (PGame.mk l r L R).Numeric) {b : Bool}
```
Really, anything more descriptive would be nice. Alternatively, you could turn these hypothesis into `def`s like in the surreal multiplication proof.